drivers/gpio/gpio_mcux.c

```diff
@@ -77,14 +114,44 @@ static int gpio_mcux_configure(const struct device *dev,
 		return -ENOTSUP;
 	}
 
+#if defined(CONFIG_SOC_SERIES_IMXRT7XX) && defined(CONFIG_SOC_FAMILY_NXP_IMXRT)
```

Comment:
We should make this dependent on `CONFIG_PINCTRL_NXP_IOCON`, rather than the SOC itself

Reply:
However, I must modify RT700 pinctrl model to adapt RT700 iopctl model. So do you mean I can create a new driver like pinctrl_lpc_iocon_1.c to adapt RT700 new model?

Reply:
The IOCON specific features in this driver still need to be made dependent on `CONFIG_PINCTRL_NXP_IOCON`